util/types/json/serdes.go

```diff
@@ -305,11 +280,7 @@ func (m *jsonObject) decode(data []byte) (err error) {
 	return
 }
 
-func (a jsonArray) getTypeCode() byte {
-	return typeCodeArray
-}
-
-func (a jsonArray) encode(buffer *bytes.Buffer) {
+func encodeJSONArray(a []JSON, buffer *bytes.Buffer) {
```

Comment:
package util/codec cannot access JSON internal fields, so I put serialize/deserialize here.
and, codec in util/codec will use these interfaces.